src/qtoolkit/io/sge.py

```diff
@@ -0,0 +1,526 @@
+                self._get_element_text(job_element, "hard resource_list.h_rt")
+            )
+
+            qjob.info = info
+
+            jobs_list.append(qjob)
+
+        return jobs_list
+
+    @staticmethod
+    def _convert_str_to_time(time_str: str | None) -> int | None:
+        if time_str is None:
+            return None
+
+        parts = time_str.split(":")
+        if len(parts) == 3:
+            hours, minutes, seconds = parts
+        elif len(parts) == 2:
+            hours, minutes = "0", parts[0]
+            seconds = parts[1]
+        elif len(parts) == 1:
+            hours, minutes, seconds = "0", "0", parts[0]
+        else:
+            raise OutputParsingError(f"Invalid time format: {time_str}")
+
+        try:
+            return int(hours) * 3600 + int(minutes) * 60 + int(seconds)
+        except ValueError:
+            raise OutputParsingError(f"Invalid time format: {time_str}")
+
+    @staticmethod
+    def _convert_memory_str(memory: str | None) -> int | None:
+        if not memory:
+            return None
+
+        match = re.match(r"([0-9]+)([a-zA-Z]*)", memory)
+        if not match:
+            raise OutputParsingError("No numbers and units parsed")
+        memory, units = match.groups()
+
+        power_labels = {"K": 0, "M": 1, "G": 2, "T": 3}
+
+        if not units:
+            units = "M"
+        elif units not in power_labels:
+            raise OutputParsingError(f"Unknown units {units}")
+        try:
+            v = int(memory)
+        except ValueError:
+            raise OutputParsingError
+
+        return v * (1024 ** power_labels[units.upper()])
+
+    _qresources_mapping = {
+        "queue_name": "queue",
+        "job_name": "job_name",
+        "account": "account",
+        "priority": "priority",
+        "output_filepath": "qout_path",
+        "error_filepath": "qerr_path",
+        "project": "group_list",
+    }
+
+    @staticmethod
+    def _convert_time_to_str(time: int | float | timedelta) -> str:
+        if not isinstance(time, timedelta):
+            time = timedelta(seconds=time)
+
+        hours, remainder = divmod(int(time.total_seconds()), 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        time_str = f"{hours}:{minutes}:{seconds}"
+        return time_str
+
+    def _convert_qresources(self, resources: QResources) -> dict:
+        header_dict = {}
+        for qr_field, sge_field in self._qresources_mapping.items():
+            val = getattr(resources, qr_field)
+            if val is not None:
+                header_dict[sge_field] = val
+
+        if resources.njobs and resources.njobs > 1:
+            header_dict["array"] = f"1-{resources.njobs}"
+
+        if resources.time_limit:
+            header_dict["walltime"] = self._convert_time_to_str(resources.time_limit)
+            header_dict["soft_walltime"] = self._convert_time_to_str(
+                resources.time_limit * 0.9
+            )
```

Comment:
What does the "soft" option does exactly here? I tried looking a bit but it is often not well documented. From [here](https://2021.help.altair.com/2021.1/AltairGridEngine/8.7.0/AdminsGuideGE.pdf) it says
> The soft real time limit (wall clock time limit) warns a job
with a catchable SIGUSER1 signal, if exceeded. After a defined time period (see notify parameter), the job is killed.

and the default for `notify` seems to be 60 seconds.
If this is the case I am convinced that an hardcoded value with a 0.9 factor would be good here.

Reply:
the soft option allows the job to exceed the allocated time limit for a bit and my colleague would actually need that option, so I would like to keep it like that.

Reply:
I might set that factor to a different value or even remove it. I lack the experience with SGE to know which approach would be best

Reply:
I set the factor to 0.99 now and would just try it like that